### PR TITLE
clear up console by not running code that we know will fail in dev

### PR DESCRIPTION
**Pull request by sndrs**
it's become pretty tricky to find errors in the console in dev.

this prevents code from running that we know cannot succeed in dev, while trying to make it plain how to re-enable it if you're working on it (specifically the service worker)

before:
screen shot 2016-01-27 at 16 19 41

after:
screen shot 2016-01-27 at 16 18 10

### CODE REVIEW

**Comment by OliverJAsh** [comment]
Hey! Could you post a full screenshot of the console beforehand so I can see the full error messages? I would like to understand what exactly is causing these errors. I don't have the same errors but that's likely because my environment is different (I'm not signed in, etc.)

I'm happy to disable the SW in dev, but if I can understand the exact issues, I will see if I can clean them up in dev without disabling the SW altogether. The SW should work in dev but it shouldn't be as noisy as it is.

I believe push notifications are right around the corner so the SW may need to be enabled in dev for that, I'm not sure.

:+1:

File: common/app/views/fragments/head.scala.html [hunk 1]
```diff
@@ -10,7 +10,9 @@
 @* stylesheet <link>s - get the stylesheets downloading ASAP *@
 @fragments.stylesheets(projectName, getContent(page).exists(_.tags.isCrossword))
 
-<link rel="prefetch" href="@Static("javascripts/app.js")">
+@if(!play.Play.isDev()) {
+    <link rel="prefetch" href="@Static("javascripts/app.js")">
+}
```
Comment (OliverJAsh) [inline common/app/views/fragments/head.scala.html]: Which console errors was this causing for you, out of curiosity?

**Comment by desbo** [comment]
what if we used a query param or cookie to enable SW in dev rather than having to flip those comments - can see that slipping into a few git commits unintentionally

**Comment by sndrs** [comment]
this is the full console 

screen shot 2016-01-27 at 17 02 17

File: common/app/views/fragments/head.scala.html (hunk 1 above)
Comment (sndrs) [inline common/app/views/fragments/head.scala.html]: it triggers one of these

screen shot 2016-01-27 at 17 04 15

**Comment by sndrs** [comment]
> what if we used a query param or cookie to enable SW in dev rather than having to flip those comments - can see that slipping into a few git commits unintentionally

yeah i wondered about that too, maybe you're right

**Comment by OliverJAsh** [comment]
Thanks. Looks like you need to be logged in to reproduce this, so I'll give that a go.

**Comment by johnduffell** [comment]
it's a great idea to clean things up in dev, I've had trouble with this too and it's difficult to get confidence that my changes are good without this. :+1: I do also have discussion errors due to the fact I don't have it set up locally which I guess are separate.

**Comment by sndrs** [comment]
@desbo updated to use `#force-sw` instead of editing the source to force it in local.

happy to see this reverted if we can sort out the errors properly, but will merge this for now if were ok with this option till then?

**Comment by desbo** [comment]
sounds good to me :+1:

**Pull request merged**